adapters/playdigo/playdigo.go

```diff
@@ -0,0 +1,164 @@
+package playdigo
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/prebid/openrtb/v20/openrtb2"
+	"github.com/prebid/prebid-server/v2/adapters"
+	"github.com/prebid/prebid-server/v2/config"
+	"github.com/prebid/prebid-server/v2/openrtb_ext"
+)
+
+type adapter struct {
+	endpoint string
+}
+
+type reqBodyExt struct {
+	PlaydigoBidderExt reqBodyExtBidder `json:"bidder"`
+}
+
+type reqBodyExtBidder struct {
+	Type        string `json:"type"`
+	PlacementID string `json:"placementId,omitempty"`
+	EndpointID  string `json:"endpointId,omitempty"`
+}
+
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
+	bidder := &adapter{
+		endpoint: config.Endpoint,
+	}
+	return bidder, nil
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var errs []error
+	var adapterRequests []*adapters.RequestData
+
+	reqCopy := *request
+	for _, imp := range request.Imp {
+		reqCopy.Imp = []openrtb2.Imp{imp}
+
+		var bidderExt adapters.ExtImpBidder
+		var playdigoExt openrtb_ext.ImpExtPlaydigo
+
+		if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
+			errs = append(errs, err)
+			continue
+		}
+		if err := json.Unmarshal(bidderExt.Bidder, &playdigoExt); err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		impExt := reqBodyExt{PlaydigoBidderExt: reqBodyExtBidder{}}
+
+		if playdigoExt.PlacementID != "" {
+			impExt.PlaydigoBidderExt.PlacementID = playdigoExt.PlacementID
+			impExt.PlaydigoBidderExt.Type = "publisher"
+		} else if playdigoExt.EndpointID != "" {
+			impExt.PlaydigoBidderExt.EndpointID = playdigoExt.EndpointID
+			impExt.PlaydigoBidderExt.Type = "network"
+		}
+
+		finalyImpExt, err := json.Marshal(impExt)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		reqCopy.Imp[0].Ext = finalyImpExt
+
+		adapterReq, err := a.makeRequest(&reqCopy)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		if adapterReq != nil {
+			adapterRequests = append(adapterRequests, adapterReq)
+		}
```

Comment:
you could directly say - `adapterRequests = append(adapterRequests, adapterReq)`